### ISSUE TITLE
no-array-method-this-argument false positive with Cypress.find

### ISSUE THREAD

**Issue by jase88** [sindresorhus/eslint-plugin-unicorn#1611]
I get false-positives for the rule **no-array-method-this-argument** if using [find from cypress](https://docs.cypress.io/api/commands/find).

Strangely this only occurs if the argument for find is a separate variable, see example:
```js
const selector = 'another-selector';
// works fine
cy.get('#selector').find('another-selector', { timeout: 2000 });
// false positive
cy.get('#selector').find(selector, { timeout: 2000 });
```

![image](https://user-images.githubusercontent.com/804836/141957680-dfeae508-6a1c-4e8c-91ad-6e6267ab8189.png)

**Comment by fisker** [comment]
We can analyze `selector`, if it's known not a `function`, ignore it.

**Comment by jase88** [comment]
It is a fixed string like in the example. What do you mean with "ignore it"? eslint-disable-next-line?

Is it necessary to analyze selector? Shouldn't it be enough to check if find() was called from an array (and not from a Cypress Chainable context)? Sorry just my naive view 🙃

**Comment by dkMorlok** [comment]
Same for me for code (second parameter is array):

```
return await this.persister.chats.find(accountId, [
	{ field: 'extId', value: values.extId },
])
```

**Comment by kidonng** [comment]
Encountered the same issue with the [`fzf`](https://github.com/ajitid/fzf-for-js) package.

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1394
